src/core/actions/unitactions/Move.java

```diff
@@ -1,45 +1,170 @@
     public boolean isFeasible(final GameState gs)
     {
-        //TODO: isFeasible this Move action
+        TreePathfinder tp = new TreePathfinder(unit.getPosition(), new StepMove(gs, unit));
+
+        if(unit.checkStatus(Types.TURN_STATUS.MOVED)) {
+            return !tp.findPathTo(destination).isEmpty();
+        }
         return false;
     }
 
     @Override
-    public boolean execute(GameState gs) {
-        //TODO Execute this Move action
+    public boolean execute(GameState gs)
+    {
+        if(isFeasible(gs)) {
+            unit.setStatus(Types.TURN_STATUS.MOVED);
+            unit.setPosition(destination.x, destination.y);
+            return true;
+        }
         return false;
     }
+
+    private class StepMove implements NeighbourProvider
+    {
+        private GameState gs;
+        private Unit unit;
+
+        StepMove(GameState curGameState, Unit movingUnit)
+        {
+            this.gs = curGameState;
+            this.unit = movingUnit;
+        }
+
+        @Override
+        // from: position from which we need neighbours
+        // costFrom: is the total move cost computed up to "from"
+        // Using this.gs, this.unit, from and costFrom, gets all the adjacent neighbours to tile in position "from"
+        public ArrayList<TreeNode> getNeighbours(Vector2d from, double costFrom) {
+            TechnologyTree techTree = gs.getTribe(unit.getTribeId()).getTechTree();
+            ArrayList<TreeNode> neighbours = new ArrayList<>();
+            Board board = gs.getBoard();
+            boolean inZoneOfControl = false;
+
+            //Check if there is an enemy unit adjacent.
+            for(Vector2d tile : from.neighborhood(1, board.getSize())) {
+                if(board.getUnitAt(tile.x, tile.y).getTribeId() != unit.getTribeId()) { inZoneOfControl = true; }
+            }
+            // Each one of the tree nodes added to "neighbours" must have a position (x,y) and also the cost of moving there from "from":
+            //  TreeNode tn = new TreeNode (vector2d pos, double stepCost)
+            // We only add nodes to neighbours if costFrom+stepCost <= total move range of this.unit
+
+            for(Vector2d tile : from.neighborhood(1, board.getSize())) {
+                Types.TERRAIN terrain = board.getTerrainAt(tile.x, tile.y);
+                double stepCost = 0.0;
+
+                //Cannot move into tiles that have not been discovered yet.
+                if(!gs.getTribe(unit.getTribeId()).isVisible(tile.x, tile.y)) { continue; }
+
+                //Unit is a water unit
+                if(unit.getType() == Types.UNIT.BOAT || unit.getType() == Types.UNIT.SHIP || unit.getType() == Types.UNIT.BATTLESHIP) {
+                    switch (terrain)
+                    {
+                        case PLAIN:
+                        case CITY:
+                        case FOREST:
+                        case VILLAGE:
+                        case MOUNTAIN:
+                            continue;
```

Comment:
In fact, this is possible. The unit disembarks. We need to create the respective function to embark() in Board.